src/rpk_entity.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2015-2021 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR
+// ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+// OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use crate::error::Error;
+use crate::{der, signed_data, DerTypeId};
+use pki_types::{SignatureVerificationAlgorithm, SubjectPublicKeyInfoDer};
+use signed_data::SubjectPublicKeyInfo;
+
+/// A Raw Public Key certificate, used for connections using raw public keys as specified in RFC7250.
+#[derive(Debug)]
+pub struct RawPublicKeyEntity<'a> {
+    inner: SubjectPublicKeyInfo<'a>,
+}
+
+impl<'a> TryFrom<&'a SubjectPublicKeyInfoDer<'a>> for RawPublicKeyEntity<'a> {
+    type Error = Error;
+
+    /// Parse the ASN.1 DER-encoded SPKI encoding of the raw public key `cert`.
+    /// Since we are parsing a raw public key, we first strip the outer sequence tag.
+    fn try_from(cert: &'a SubjectPublicKeyInfoDer<'a>) -> Result<Self, Self::Error> {
+        let input = untrusted::Input::from(cert.as_ref());
+        let spki = input.read_all(Error::TrailingData(DerTypeId::Certificate), |reader| {
```

Comment:
```suggestion
        let spki = input.read_all(Error::TrailingData(DerTypeId::SubjectPublicKeyInfo), |reader| {
```
introduce a new error for this

Reply:
Could you explain why we need a new error? In `Cert` `TrailingData` is also used for reading the certificate.

Reply:
Apologies, I mean a new `DerTypeId` variant. This names which type we were trying to decode when we found excess bytes.